draft-ietf-jsonpath-base.md

```diff
@@ -573,6 +573,13 @@ root-selector  = "$"
 The Argument — the root JSON value — becomes the root node, which is
 addressed by the root selector `$`.
 
+#### Examples
+{: unnumbered}
+
+| JSON | Query | Result | Comment|
```

Comment:
Shouldn't we give the value in addition to the path for the result?
This is a bit tautological at the moment.

Reply:
Yeah, that's probably more friendly to the users. I'll do it.

Reply:
Drawbacks with Markdown tables are:
* Column sizes are not explicitly adjustable; but implicitly by Header length.
* Multiline codeblocks are not supported inside. HTML `<br>`s are necessary here.

Especially rendered plain text leaves only few horizontal space.

So I suggest:

* Widening / Shortening Heading-text where appropriate.
* Minimize number of columns.
* Use short comments only.

| JSON document  | Query | Value | Result | Comment| 
| --- | --- | --- | --- | --- |
| `{ "o": {"key":7}}` | `$..o` | `{"key":7}` | `$['o']` | Child node  |
|  | `$..['key']` | 7 | `$['o']['key']` | Descendent node  |
|  | `$..*`  | `{"key":7}`<br> 7  | `$['o']`<br>`$['o']['key']` |All descendents  | 
|  | `$..[*]`  | `{"key":7}`<br> 7  | `$['o']`<br>`$['o']['key']` |All descendents  | 

Maybe with that space restrictions we might consider to show either result values or normalized pathes.
Another possibility would be to write the JSON before the table; then restricting all queries to that document.

Reply:
Not commenting on the content of the table:  The RPC (RFC production center) often re-writes tables that have become unwieldy into definition lists.

(Note that most of the limitations you perceive here are those of RFCXML; markdown is just a way to input that RFCXML.)

The table here could be made more readable by leaving out the common value "JSON document" and just mentioning that before the start of the table.

Reply:
Well ... then we have ...

### descendant-selector

```json
{ 
  "o": {
    "key":7
  }
}
```
| Query | Value | Result | Comment| 
| --- | --- | --- | --- |
|`$..o` | `{"key":7}` | `$['o']` | Child node  |
| `$..['key']` | 7 | `$['o']['key']` | Descendent node  |
| `$..*`  | `{"key":7}`<br> 7  | `$['o']`<br>`$['o']['key']` |All descendents  | 
| `$..[*]`  | `{"key":7}`<br> 7  | `$['o']`<br>`$['o']['key']` |All descendents  |

Reply:
Please see 5df7f520de3dbce58f289636779233017cc340d5. The text rendering is still far from perfect. I don't know how to adjust column widths reliably.